src/qibolab/instruments/qm/config.py

```diff
@@ -50,18 +50,26 @@ def register_port(self, port):
             is_octave = isinstance(port, (OctaveOutput, OctaveInput))
             controllers = self.octaves if is_octave else self.controllers
             if port.device not in controllers:
-                controllers[port.device] = {}
-                if not is_octave:
-                    controllers[port.device]["analog_inputs"] = DEFAULT_INPUTS
+                if is_octave:
+                    controllers[port.device] = {}
+                else:
+                    controllers[port.device] = {
+                        "analog_inputs": DEFAULT_INPUTS,
+                        "digital_outputs": {},
+                    }
 
             device = controllers[port.device]
             if port.key in device:
                 device[port.key].update(port.config)
             else:
                 device[port.key] = port.config
```

Comment:
Of course not in this PR, but maybe `device` could be a `defaultdict`? It would remove the need for the `if`.

Reply:
Maybe yes, with a small concern because this is part of the `config` that is sent to QM libraries, so I am not sure if using `defaultdict` over just `dict` will create any issues depending on what manipulations they are doing.

Anyway, as you said I wouldn't do this here. In fact I would like to get rid of the `QMConfig` class from the driver at some point (replace `QMConfig.register_*` with smaller dataclasses/`asdict`), but I haven't tried yet.